source/js/components/newsletter-signup/organisms/with-submission-logic.jsx

```diff
@@ -1,16 +1,25 @@
 import React from "react";
+import PropTypes from "prop-types";
 import { getText } from "../../petition/locales";
 
 /**
  * Higher-order component that handles form submission logic and validation
  * for newsletter signup forms.
  */
 function withSubmissionLogic(WrappedComponent) {
-  return class extends React.Component {
+  class WithSubmissionLogicComponent extends React.Component {
```

Comment:
Form submission logic and functionalities are pretty much based on `join.jsx` but with a cleaner implementation.